javaslang/src/main/java/javaslang/Value.java

```diff
@@ -1021,6 +1013,26 @@ default CharSeq toCharSeq() {
     @Override
     String toString();
 
+    /**
+     * Tests equality of two objects assuming one, or both might be instances of Value.
+     *
+     * If one of a or b is Value instance congruence is tested.
+     * If neither is Objects{@link Objects#equals} is being invoked for two.
+     *
+     * @param a first object to test
+     * @param b fecond object to test
+     * @return true, if this equals o according to the rules defined above, otherwise false.
+     */
+    static boolean eq(Object a, Object b) {
```

Comment:
Yes, I think for now we only need a package-private `Sets` class (similar to `Maps`), which contains one method:

```java
static boolean equals(Set<?> self, Object o) {
    if (o == self) {
        return true;
    } else if (o instanceof Set) {
        final Set<?> that = (Set<?>) that;
        return self.forAll(that::contains); // PROBLEM: SEE BELOW
    } else {
        return false;
    }
}
```

The `Set` interface needs to `@Override` the `equals` method. This should contain some javadoc about Set equality (all Set impls share the same equals method).

All Set implementations need to override the equals method like this:

```java
@Override
public boolean equals(Object o) {
    return Sets.equals(this, o);
}
```

We only have one problem to solve in order to make the code above work: `contains()` currently takes a `T` instead of an `Object`. Maybe we need to change the signature of `Set#contains`...